docs/form.mdx

```diff
@@ -0,0 +1,180 @@
+    <UppyCdnExample>
+      {`
+        import { Uppy, Form } from "{{UPPY_JS_URL}}"
+        const uppy = new Uppy()
+        uppy.use(Form, {
+          // Options
+        })
+      `}
+    </UppyCdnExample>
+  </TabItem>
+</Tabs>
+
+## Use
+
+```js title="app.js"
+import Uppy from '@uppy/core';
+import Form from '@uppy/form';
+import DragDrop from '@uppy/drag-drop';
+
+import '@uppy/core/dist/style.min.css';
+import '@uppy/drag-drop/dist/style.min.css';
+
+new Uppy().use(Form, {
+	target: '#my-form',
+});
+```
+
+```html title="index.html"
+<form id="my-form" action="/submit" method="get">
+	<label for="name">Enter your name: </label>
+	<input type="text" name="name" required />
+
+	<label for="dob">Date of birth: </label>
+	<input type="date" name="dob" />
+
+	<input type="submit" value="Save" />
+</form>
+```
+
+By default the code above will:
+
+1. Extract meta data from the form `#my-form` and send it with the Uppy upload.
+   Those fields will then be added to Uppy meta data state (`uppy.state.meta`)
+   and each file’s meta, and appended as (meta)data to the upload in an object
+   with `[file input name attribute]` -> `[file input value]` key/values.
+2. When Uppy completes upload/processing, it will add an
+   `<input type="hidden" name="uppyResult">` with the stringified upload result
+   object back to the form.
+
+:::note
+
+You can disable both of these features, see options below.
+
+:::
+
+:::tip
+
+`@uppy/form` can also start Uppy upload automatically once the form is
+submitted, and even submit the form after the upload is complete. This is off by
+default. See [`triggerUploadOnSubmit`](#triggerUploadOnSubmit) and
+[`submitOnSuccess`](#submitOnSuccess) options below for details.
+
+:::
+
+## API
+
+### Options
+
+#### `id`
+
+A unique identifier for this plugin (`string`, default: `'Form'`).
+
+#### `target`
+
+DOM element or CSS selector for the form element (`string` or `Element`,
+default: `null`)
+
+This is required for the plugin to work.
+
+#### `resultName`
+
+The `name` attribute for the `<input type="hidden">` where the result will be
+added (`string`, default: `uppyResult`)
+
+#### `getMetaFromForm`
+
+Configures whether to extract metadata from the form (`boolean`, default:
+`true`)
+
+When set to true, the `Form` plugin will extract all fields from a `<form>`
+element before upload begins. Those fields will then be added to Uppy meta data
+state (`uppy.state.meta`) and each file’s meta, and appended as (meta)data to
+the upload in an object with `[file input name attribute]` ->
+`[file input value]` key/values.
+
+#### `addResultToForm`
+
+Configures whether to add upload/encoding results back to the form in an
+`<input name="uppyResult" type="hidden">` element (`boolean`, default: `true`)
+
+#### `triggerUploadOnSubmit`
+
+Configures whether to start the upload when the form is submitted (`boolean`,
+default: `false`)
+
+When the user presses a submit button, this will prevent form submission, and
```

Comment:
```suggestion
When the user presses a submit button, type <kbd>Enter</kbd>, or does any similar action that would normally trigger the form submission, this will prevent the form submission, and
```